Components/MineSharp.Protocol/Packets/Clientbound/Play/EntitySoundEffectPacket.cs

```diff
@@ -0,0 +1,70 @@
+﻿using MineSharp.Core.Common;
+using MineSharp.Core.Serialization;
+using MineSharp.Data;
+using MineSharp.Data.Protocol;
+
+namespace MineSharp.Protocol.Packets.Clientbound.Play;
+#pragma warning disable CS1591
+public sealed record EntitySoundEffectPacket(
```

Comment:
EntitySoundEffect packet changed in 1.19 and again in 1.19.3

Reply:
With all that differences between versions how should we deal with that?

1. We could make the packet class able to handle all the versions. But that often means that the paket gets fields that are not used in other versions. So we need to make them nullable. And also need to document when (for which version) these nullable fields are used.

2. Or alternatively we could create a implementation of that packet class for every version that changed something in that packet. This makes using the packet classes more straitforward since you really get all the fields it hat. But that would mean that for different versions difference PacketHandlers are required (and need to be registered with `On<PACKET>`).

I like option 2 more. What do you think?

Reply:
Yeah, i've been struggeling with this problem ever since I started this project...
I'd also go with Option B. Not too long ago I did something similar with [SystemChatMessagePacket](https://github.com/psu-de/MineSharp/blob/499c76703d8cbd715099ac1e5cc06114de7e125a/Components/MineSharp.Protocol/Packets/Clientbound/Play/SystemChatMessagePacket.cs). I used an abstract base class so the different versions don't have to be registered themselves.

Reply:
btw. don't worry if you don't want to bother implementing all different packet versions. I'll do it at some point when I'm bored...

Reply:
I think that we need an even more abstract/universal approach than that currently used in [SystemChatMessagePacket](https://github.com/psu-de/MineSharp/blob/499c76703d8cbd715099ac1e5cc06114de7e125a/Components/MineSharp.Protocol/Packets/Clientbound/Play/SystemChatMessagePacket.cs). A system that is consistent between all packets.
I mean I would also do this with using inheritance of the base packet class. But all the possible version specific sub classes need to implement a common interface and need to be registered in a lookup table (per packet).

We should probably create an issue regarding that topic.

Reply:
Hi,
sorry it took a while....

I've updated [this Issue](https://github.com/psu-de/MineSharp/issues/68#issue-2428145448), and tried to write down my thoughts.